### PR TITLE
feat(perf-issues): Improve legibility of duration text within autogrouped spans & affected spans

### PR DESCRIPTION
**Pull request by 0Calories**
This PR improves the legibility of the duration text within autogrouped spans & affected spans by introducing a backplate behind the text. 

The colours being used and other styling is tentative, need approval from Dora first! Also to note: There are some custom colours being used here that are not included in the theme. We may need dark mode colour alternatives, or to bring these colours into the themes as well.

For autogroups, this visual change will also be present on Event Details, not just on Issue Details.

## Affected Spans Before & After
image
image

## Missing Instrumentation (Gap) Spans Before & After
image
image

## Autogrouped Spans Before & After
image
image
### Another Example
image
image

### CODE REVIEW

File: static/app/components/events/interfaces/spans/spanBar.tsx [hunk 1]
```diff
@@ -898,7 +895,7 @@ class SpanBar extends Component<SpanBarProps, SpanBarState> {
     errors: TraceError[] | null;
     transactions: QuickTraceEvent[] | null;
   }) {
-    const {span, spanBarColor, spanBarHatch, spanNumber} = this.props;
+    const {span, spanBarColor, spanBarType: spanBarType, spanNumber} = this.props;
```
Comment (gggritso) [inline static/app/components/events/interfaces/spans/spanBar.tsx:898]: Stray refactor?
```suggestion
    const {span, spanBarColor, spanBarType, spanNumber} = this.props;
```

File: static/app/components/performance/waterfall/utils.tsx [hunk 2]
```diff
@@ -73,24 +73,25 @@ export const getDurationPillAlignment = ({
...
+  spanBarType?: SpanBarType;
 }) => {
   if (durationDisplay === 'inset') {
-    return `color: ${
-      spanBarHatch && spanBarHatch === SpanBarHatch.gap ? theme.gray300 : theme.white
-    };`;
+    const {alternate, insetTextColour} = getSpanBarColours(theme, spanBarType);
+    return `background: ${alternate}; color: ${insetTextColour};`;
+    // const isGapSpan = spanBarType && spanBarType === SpanBarType.GAP;
+    // return `color: ${isGapSpan ? theme.gray300 : theme.white};`;
```
Comment (gggritso) [inline static/app/components/performance/waterfall/utils.tsx:91]: Also stray?
```suggestion
```

File: static/app/components/performance/waterfall/utils.tsx [hunk 3]
```diff
@@ -25,9 +25,9 @@ export const getBackgroundColor = ({
   return theme.background;
 };
 
-export function getHatchPattern(spanBarHatch: SpanBarHatch | undefined) {
-  if (spanBarHatch) {
-    const {primary, alternate} = SPAN_HATCH_TYPE_COLOURS[spanBarHatch];
+export function getHatchPattern(spanBarType: SpanBarType | undefined, theme: Theme) {
+  if (spanBarType) {
+    const {primary, alternate} = getSpanBarColours(theme, spanBarType);
```
Comment (gggritso) [inline static/app/components/performance/waterfall/utils.tsx:30]: TypeScript makes a mistake impossible, but it's still a little strange that the order of arguments for `getHatchPattern` and `getSpanBarColours` is type, theme for the former and theme, type for the latter

**Review by gggritso — APPROVED** [review]
Very nice! 🦓

File: static/app/components/events/interfaces/spans/spanBar.tsx (hunk 1 above)
Comment (0Calories) [reply static/app/components/events/interfaces/spans/spanBar.tsx:898]: Yep, thanks for catching this!

File: static/app/components/performance/waterfall/utils.tsx (hunk 2 above)
Comment (0Calories) [reply static/app/components/performance/waterfall/utils.tsx]: yessir

File: static/app/components/performance/waterfall/utils.tsx (hunk 3 above)
Comment (gggritso) [reply static/app/components/performance/waterfall/utils.tsx:30]: Ah, makes sense! Up to you

**Review by doralchan — APPROVED** [review]
Shipit

**Pull request merged**